assimilation_code/location/threed_cartesian/location_mod.f90

```diff
@@ -1222,8 +1422,8 @@ end subroutine find_box_ranges
 
 !----------------------------------------------------------------------------
 
-subroutine find_nearest(gc, base_loc, loc_list, nearest, rc)
- type(get_close_type), intent(in), target  :: gc
+subroutine find_nearest(box, base_loc, loc_list, nearest, rc)
```

Comment:
find_nearest() is a public routine so it can't have a box_type as the first arg.  it needs to be passed a get_close_type.   this routine doesn't use maxdist at all - it returns the single nearest location to the target.  it should be ok to simply use bt=1 to get the first box_type from the gc.

Reply:
This all makes sense to me.  After looking through the DART code, I realized this subroutine is not called anywhere.  I bet that is why I never had any errors come up previously when running the code.

Also sorry, but this is my first time making any type of contribution to a public code package. To correct these bugs, do you want me to fix my own local copy and push it into the pull request?

Reply:
Also I have a few more questions:  
1.) What is the best strategy to make sure this branch is up to date with the main code?  Should I update my main branch and use that to rebase this current branch?

2.) Since opening this pull request I created a new branch that is capable of making horizontal/vertical localization radii different for CM1.  Should I complete this pull request first, and once complete open a new one for those updates?

Reply:
hi jon.  yes, please fix your local copy and push the changes.  we'll see them here and can review what's been addressed.

i'll let helen reply to your git questions.  (i'm a novice at git but i can answer any subversion questions you might have!)

i would definitely finish this pull request and make the horizontal/vertical localization changes a different pull request.  multiple smaller pull requests are easier to review and test, and help to ensure lurking bugs don't creep in with huge changes that don't get caught for a long time.